src/rai_bench/rai_bench/tool_calling_agent/mocked_tools.py

```
@@ -265,45 +268,163 @@ def _run(self, msg_type: str) -> str:
+        ValueError
+            If service_type is not found in custom_models or if Pydantic
+            validation fails.
+        """
+        if service_type not in self.custom_models:
+            raise ValueError(f"Service type: {service_type} is invalid custom type")
+
+        model = self.custom_models[service_type]
+        try:
+            model.model_validate(args)
+        except ValidationError as e:
+            raise ValueError(f"Pydantic validation failed: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)